packages/forklift-console-plugin/webpack.config.ts

```diff
@@ -39,7 +39,7 @@ const config: WebpackConfiguration & {
   module: {
     rules: [
       {
-        test: /\.(jsx?|tsx?)$/,
+        test: /\.(tsx?)$/,
```

Comment:
Does dropping js from ts-loader interfere with the `tsconfig.json` option set  `"allowJs": true`?

Full effective config:
```
[sdickers@twilight forklift-console-plugin]$ npx tsc --showConfig | jq '.'
{
  "compilerOptions": {
    "allowJs": true,
    "strict": false,
    "allowSyntheticDefaultImports": true,
    "resolveJsonModule": true,
    "skipLibCheck": true,
    "noUnusedLocals": true,
    "target": "es2020",
    "module": "es2020",
    "moduleResolution": "node",
    "sourceMap": true,
    "declaration": true,
    "declarationMap": true,
    "jsx": "react",
    "lib": [
      "dom",
      "es2020"
    ],
    "esModuleInterop": true,
    "outDir": "./dist",
    "paths": {
      "src/*": [
        "./src/*"
      ],
      "@kubev2v/legacy/*": [
        "../legacy/src/*"
      ],
      "@kubev2v/common/*": [
        "../common/src/*"
      ],
      "@kubev2v/mocks/*": [
        "../mocks/src/*"
      ],
      "@kubev2v/webpack": [
        "../webpack/src"
      ],
      "@kubev2v/types": [
        "../types/src"
      ]
    }
  },
  "references": [
    {
      "path": "../common"
    },
    {
      "path": "../types"
    },
    {
      "path": "../legacy"
    }
  ],
  "files": [
    "./src/__mocks__/console_components.tsx",
    "./src/__mocks__/dummy.ts",
    "./src/__mocks__/envvars.ts",
    "./src/__mocks__/react-i18next.tsx",
    "./src/components/cells/StatusCell.tsx",
    "./src/components/cells/TextWithIcon.tsx",
    "./src/components/cells/utils.tsx",
    "./src/components/empty-states/ForkliftEmptyState.tsx",
    "./src/components/empty-states/images/globals.d.ts",
    "./src/components/mappings/MappingPage.tsx",
    "./src/components/mappings/MappingRow.tsx",
    "./src/components/mappings/data.ts",
    "./src/components/mappings/mappingActions.tsx",
    "./src/components/page/ManageColumnsToolbar.tsx",
    "./src/components/page/StandardPage.tsx",
    "./src/components/page/__tests__/StandardPage.test.tsx",
    "./src/mock-console-extension/dynamic-plugin.ts",
    "./src/mock-console-extension/index.tsx",
    "./src/mock-console-extension/logMockRequest.ts",
    "./src/mock-console-extension/setupBrowserWorker.tsx",
    "./src/modules/Hooks/dynamic-plugin.ts",
    "./src/modules/Migrations/dynamic-plugin.ts",
    "./src/modules/NetworkMaps/EmptyStateNetworkMaps.tsx",
    "./src/modules/NetworkMaps/NetworkMappingRow.tsx",
    "./src/modules/NetworkMaps/NetworkMappingsPage.tsx",
    "./src/modules/NetworkMaps/NetworkMappingsWrapper.tsx",
    "./src/modules/NetworkMaps/UseNetworkMappingActions.tsx",
    "./src/modules/NetworkMaps/dataForNetwork.ts",
    "./src/modules/NetworkMaps/dynamic-plugin.ts",
    "./src/modules/NetworkMaps/mappingActions.tsx",
    "./src/modules/NetworkMaps/__tests__/MappingRow.test.tsx",
    "./src/modules/NetworkMaps/__tests__/mergeData.test.ts",
    "./src/modules/Plans/EmptyStatePlans.tsx",
    "./src/modules/Plans/PlanRow.tsx",
    "./src/modules/Plans/PlanWizardWrapper.tsx",
    "./src/modules/Plans/PlansPage.tsx",
    "./src/modules/Plans/PlansWrapper.tsx",
    "./src/modules/Plans/UsePlanActions.tsx",
    "./src/modules/Plans/VMMigrationDetailsWrapper.tsx",
    "./src/modules/Plans/data.ts",
    "./src/modules/Plans/dynamic-plugin.ts",
    "./src/modules/Plans/planActions.tsx",
    "./src/modules/Plans/__tests__/PlanRow.test.tsx",
    "./src/modules/Plans/__tests__/mergeData.test.ts",
    "./src/modules/Providers/AddProviderButton.tsx",
    "./src/modules/Providers/EmptyStateProviders.tsx",
    "./src/modules/Providers/HostsPageWrapper.tsx",
    "./src/modules/Providers/ProviderRow.tsx",
    "./src/modules/Providers/ProvidersPage.tsx",
    "./src/modules/Providers/ProvidersWrapper.tsx",
    "./src/modules/Providers/UseMergedProviders.ts",
    "./src/modules/Providers/data.ts",
    "./src/modules/Providers/dynamic-plugin.ts",
    "./src/modules/Providers/index.ts",
    "./src/modules/Providers/providerActions.tsx",
    "./src/modules/Providers/ProviderRow/HostCell.tsx",
    "./src/modules/Providers/ProviderRow/ProviderLink.tsx",
    "./src/modules/Providers/ProviderRow/StatusCell.tsx",
    "./src/modules/Providers/ProviderRow/TextCell.tsx",
    "./src/modules/Providers/ProviderRow/TypeCell.tsx",
    "./src/modules/Providers/ProviderRow/consts.tsx",
    "./src/modules/Providers/ProviderRow/types.ts",
    "./src/modules/Providers/__tests__/ProviderRow.test.tsx",
    "./src/modules/Providers/__tests__/data.test.ts",
    "./src/modules/StorageMaps/EmptyStateStorageMaps.tsx",
    "./src/modules/StorageMaps/StorageMappingRow.tsx",
    "./src/modules/StorageMaps/StorageMappingsPage.tsx",
    "./src/modules/StorageMaps/StorageMappingsWrapper.tsx",
    "./src/modules/StorageMaps/UseStorageMappingActions.tsx",
    "./src/modules/StorageMaps/dataForStorage.ts",
    "./src/modules/StorageMaps/dynamic-plugin.ts",
    "./src/modules/StorageMaps/mappingActions.tsx",
    "./src/modules/StorageMaps/__tests__/MappingRow.test.tsx",
    "./src/modules/StorageMaps/__tests__/mergeData.test.ts",
    "./src/utils/constants.ts",
    "./src/utils/enums.ts",
    "./src/utils/fetch.ts",
    "./src/utils/i18n.ts",
    "./src/utils/resources.ts",
    "./src/utils/types.ts",
    "./src/utils/__tests__/resources.test.ts",
    "./plugin-metadata.ts",
    "./plugin-extensions.ts"
  ],
  "include": [
    "./src/**/*",
    "./plugin-metadata.ts",
    "./plugin-extensions.ts"
  ],
  "exclude": [
    "./dist"
  ]
}
```

Reply:
why do you think it interfere? 
can you check if we can safely disable allows? 
although IMHO we don't need to, WDYT ?

Reply:
IMHO we can keep processing `js` files. Sometimes we do use them and we'll waste time debugging why they are not picked by the build.

Reply:
@rszwajko to be clear, you are ok with leaving `allowJs": true` while the webpack rule don't send them to the ts compiler ?

Reply:
I'd rather keep both `allowJs: true` and ` (jsx?|tsx?)$`. Not much gain here and may cause some troubles in the future. But if we want to be strict TS-only then probably we should remove both.

Reply:
there are two rules to handle .js files, one need to go
I prefere to remove the first 9ne, because IMHO it's unecceary,
I can try to remove the other rule to handle .js files from the end
but I will need to test that solution first

p.s.
having two rules for .js files is problematic, to see the effect, take this PR and re enable the first rule to handle also .js files

Reply:
OK. I've missed the second rule - I've thought we want to switch to TS-only setup. 
Dual setup seems usually implemented by using ts-loader only - i.e. see https://webpack.js.org/guides/typescript/

Reply:
> I've thought we want to switch to TS-only setup

the PR description describe the issues
one - monorepo / path conflict
two - duplicate rules for .js and .svg files